src/app/frontend/replicasetdetail/replicasetdetail_controller.js

```diff
@@ -20,12 +20,38 @@
 export default class ReplicaSetDetailController {
   /**
    * @param {!backendApi.ReplicaSetDetail} replicaSetDetail
+   * @param {!backendApi.Events} replicaSetEvents
    * @ngInject
    */
-  constructor(replicaSetDetail) {
-    /**
-     * @export {!backendApi.ReplicaSetDetail}
-     */
+  constructor(replicaSetDetail, replicaSetEvents) {
+    /** @export {!backendApi.ReplicaSetDetail} */
     this.replicaSetDetail = replicaSetDetail;
+
+    /** @export {!backendApi.Events} */
+    this.replicaSetEvents = replicaSetEvents;
+
+    /** @const @export {!Array<string>} */
+    this.eventTypeFilter = ['All', 'Warning'];
+
+    /** @export {string} */
+    this.eventType = this.eventTypeFilter[0];
+
+    /** @const @export {!Array<string>} */
+    this.eventSourceFilter = ['All', 'User', 'System'];
+
+    /** @export {string} */
+    this.eventSource = this.eventSourceFilter[0];
+  }
+
+  /**
+   * Formats labels object to readable string.
+   * @param {!Object} object
+   * @return {string}
+   * @export
+   */
+  formatLabelString(object) {
```

Comment:
I can extract this to export function then import in second controller and assign to fn variable. Is it ok or do you have something else in mid?

```javascript
import {formatLabelString} from 'replicasetdetail/replicasetdetail_controller';
.... in controller
this.formatLabelString = formatLabelString;
```